scripts/cleanup_old_vms.py

```diff
@@ -71,19 +79,48 @@ def get_vm_config_modified_time(name, vm_name, datastore_url, provider_key):
         return False
 
 
-def process_provider_vms(provider_key, provider_type, matchers, delta, vms_to_delete):
+def list_provider_vms(provider_key):
+    try:
+        provider = get_mgmt(provider_key)
+        vm_list = provider.list_vm()
+        provider_type = cfme_data.get("management_systems", {})[provider_key].get('type', None)
+        now = datetime.datetime.now()
+        for vm_name in vm_list:
+            with open(args.outfile, 'a+') as report:
+                creation = provider.vm_creation_time(vm_name)
+                message = "{provider_type}:{provider}  {instance} " \
+                          "\t {time} \t {instance_type} \t {instance_status}\n\n" \
+                          "".format(provider_type=provider_type,
+                                    provider=provider_key,
+                                    instance=vm_name,
+                                    time=(now - creation),
+                                    instance_type=provider.vm_type(vm_name),
+                                    instance_status=provider.vm_status(vm_name))
+                print(message)
+                report.write(message)
+    except Exception as e:
+        logger.error('failed to list vms from provider {}'.format(provider_key))
+        logger.exception(e)
+
+
+def process_provider_vms(provider_key, provider_type, matchers,
+                         delta, vms_to_delete, list_vms=None):
     with lock:
         print('{} processing'.format(provider_key))
     try:
         now = datetime.datetime.now()
         with lock:
             # Known conf issue :)
             provider = get_mgmt(provider_key)
-        for vm_name in provider.list_vm():
-            if not match(matchers, vm_name):
-                continue
+        vm_list = provider.list_vm()
```

Comment:
You end up calling provider.list_vm() twice when `list_vms` is true. Given the scope of `list_provider_vms` could you pass the list instead of calling it again? If you see list_provider_vm() being used more generally in the future would probably be better to leave the call there as well.